phase-4-explainer.md

```diff
@@ -0,0 +1,175 @@
+accessing the computed accessibility tree will have the least direct impact on users.
+In the spirit of the [Priority of Constituencies](https://www.w3.org/TR/html-design-principles/#priority-of-constituencies),
+it makes sense to tackle this work last.
+
+
+#### ComputedAccessibleNode
+It is currently possible to check accessible attributes for a DOM element by using the `getAttribute` method:
+```js
+element.hasAttribute("aria-labelledby");
+element.getAttribute("aria-labelledby");
+```
+However, determining which accessible properties apply to an element can be cumbersome, as there can be multiple
+Accessing the computed accessible properties of a DOM element will be designed to mimic the behaviour of `getComputedStyle()`, but implemented as an asynchronous Promise-based API. A `ComputedAccessibleNode` can be requested for a specific element from the window:
+
+```html
+<button id="composeBtn">Compose</button>
+```
+```js
+var composeBtn = document.getElementById('composeBtn');
+window.getComputedAccessibleNode(element).then(
+  function(computedAccessibleNode) {
+    console.log(computedAccessibleNode.role);  // resolved!
+  },
+  function(error) {
+    console.log(error);  // rejected, but why?
+  });
+```
+
+The rejection case of the `Promise` is discussed in [Open Questions](#open-questions-rejection-case). Alternatively, we can use `await`:
+
+```js
+var computedAccessibleNode = await window.getComputedAccessibleNode(element);
+console.log(computedAccessibleNode.role);  // "button".
+```
+We can now access all of the computed properties exposed from one object, without relying on a leaky abstraction. This is distinct from [Phase 1](#phase-1-modifying-accessible-properties) as it would return the computed role, `"button"`, rather than `null`.  Exactly what properties a `ComputedAccessibleNode` should expose is discussed in in [Open Questions](#open-questions-accessible-properties).
+
+#### Accessibility Tree Snapshots
+`ComputedAccessibleNode`s should represent a consistent snapshot of the Accessibility Tree.
+
+#### Open Questions
+There are a number of open questions about Phase 4 that need to be discussed. This list is by no means exhaustive, but should provide a good platform for beginning discussion on the specifics of the public API design.
+
+##### 1. Accessible Properties
+The first question to answer is what properties are we expecting to expose via this API, and how? A lot of the properties are fairly self explanatory, such as `role` and `name` returning a string, or `colCount` returning an integer etc. Other attributes are not so straightforward, such as `checked`. DOM elements expose a `checked` property on all input types, which can take either of the values `true` or `false`. . Other attributes are not so straightforward, such as `checked`. Although checkboxes can only be in one of two states, checked or unchecked upon a form submission, they can take on a third visual state - `indeterminate`.
```

Comment:
Get some more linebreaks in here :) Also, you seem to repeat the checked point twice.

Reply:
Done.